app/src/components/adminConsole/currentCup/CurrentCupExportSnaps.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useCallback } from "react";
+import { CurrentCupOptionsButton } from "../AdminConsoleStyles";
+import getSnaps from "../../../firebase/snaps/GetSnaps";
+import { snapsToCsvDownload } from "../csvManager";
+
+const LOADING = "loading";
+const IDLE = "idle";
+const ERROR = "error";
+
+const CurrentCupExportSnaps = ({ cup }) => {
+    const [error, setError] = useState<string>("");
+
+    let buttonText = `Export Snaps`;
+    if (error) {
+        buttonText += "- " + error;
```

Comment:
I don't like this pattern - it'll cause the layout to jump around as extra text is added.

Could we adapt something like `SubmissionBoxErrorDisplay` for use in displaying errors in this component?

Reply:
Okay cool.